iree-prof-tools/graph-util.cc

```diff
@@ -254,16 +328,26 @@ void AddResourceAsOutputOrIncomingEdge(IREE::Stream::CmdDispatchOp op,
 
   CHECK(access & ACCESS_READ);
 
-  // First, find the last node which has an output of the same resource id.
+  // First, find the last node(s) which is not in the same stream.cmd.concurrent
+  // and has an output matched with the resource. Note that they could be more
+  // than one if last nodes are in a stream.cmd.concurrent (which is different
+  // than that of this node).
   // TODO(byungchul): Consider the control flow, e.g. scf or cf ops.
+  const GraphNode* matched = nullptr;
   for (auto it = graph.nodes.rbegin(); it != graph.nodes.rend(); ++it) {
```

Comment:
There might be a way to use the MLIR use-def chain. Instead of doing a plain reverse traversal over the graph. (Let's find a solution later.)